zip2john: add page (#9046)
tldr-pages/tldr: pages/common/zip2john.md

# zip2john

> A tool to extract password hashes from zip files for use with John the Ripper password cracker.

> > This is a utility tool usually installed as part of the John the Ripper installation.

Reviewer: please add a link to this page

Reviewer 2: ```suggestion
> This is a utility tool usually installed as part of the John the Ripper installation.
> More information: <https://github.com/openwall/john>.
```
I found out that this tool is present under `john`. References [1](https://www.openwall.com/john/), [2](https://secnhack.in/crack-zip-files-password-using-john-the-ripper/).

Or we can use 
```suggestion
> This is a utility tool usually installed as part of the John the Ripper installation.
> More information: <https://www.openwall.com/john/>.
```
to align with the other page.
https://github.com/tldr-pages/tldr/blob/d144d74b35cd702763c1883c100f620c43ad8aa1/pages/common/john.md?plain=1#L4

> - Extract the password hash from an archive, will list all files in the archive:

Reviewer: ```suggestion
- Extract the password hash from an archive, listing all files in the archive:
```

# zip2john

> A tool to extract password hashes from zip files for use with John the Ripper password cracker.
> This is a utility tool usually installed as part of the John the Ripper installation.
> More information: <https://www.openwall.com/john/>.

- Extract the password hash from an archive, listing all files in the archive:

`zip2john {{path/to/file.zip}}`

> - Extract the password hash using [o]nly the specified compressed file:

Reviewer: Oh, I think `the` is not correct here
```suggestion
- Extract the password hash using [o]nly a specific compressed file:
```

> - Extract the password has using [o]nly the specified file within the archive:

Reviewer 2: ```suggestion
- Extract the password hash using [o]nly the specified file within the archive:
```

> `zip2john -o {{compressed_file_in_zip}} {{path/to/file.zip}}`

Reviewer: Perhaps this better
```suggestion
`zip2john -o {{compressed_filename}} {{path/to/file.zip}}`
```

Author: I agree it seems less than ideal.  Since archives can contain a folder structure I didn't want to confuse "path" with the local path and wanted to point out it's the file within the archive folder structure.  Perhaps `{{archive/path/to/file}}`?  I welcome suggestions to make it clearer.

Reviewer: Based on `zgrep` page
```suggestion
`zip2john -o {{path/to/compressed_file}} {{path/to/file.zip}}`
```

> - Output the extracted hash into a file so that it can be used with John the Ripper:

Reviewer: Instead of this example why not a direct pipe to the John the Ripper app?

Author: Since there are many different formats, typical process is to read the first few characters of the hash to determine encryption format which will determine what option to pass to the John the Ripper app in the `--format=` flag.  I figured that detail is more within John the Ripper domain of instruction.

Reviewer: ```suggestion
- Extract the password hash from a compressed file to a specific file (for use with John the Ripper):
```

> `zip2john -o {{compressed_file_in_zip}} {{path/to/file.zip}} > {{file.hash}}`

Reviewer: To match the example above
```suggestion
`zip2john -o {{path/to/compressed_file}} {{path/to/file.zip}} > {{file.hash}}`
```
